1-JavaScript/Week3/homework/ex5-shoppingCartPure.js

```diff
@@ -16,8 +16,14 @@ it pure. Do the following:
 5. Confirm that you function passes the provided unit tests.
 ------------------------------------------------------------------------------*/
 // ! Function under test
-function addToShoppingCart(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function addToShoppingCart(arrayShoppingList, groceryItem) {
+  const newShoppingCart = [...arrayShoppingList];
+  newShoppingCart.push(groceryItem);
+  if (newShoppingCart.length > 3) {
+    newShoppingCart.shift();
```

Comment:
@Ozgur2096  You can simplify your solution here as the following:
 ```suggestion
 return [... arrayShoppingList, groceryItem].slice(-3);
```
This one line will do the same thing that you are doing. You can merge `groceryItem`  & `arrayShoppingList ` then do slice(-3)  that will make sure you only have 3 items

Here's more details about es6 spread operator https://www.w3schools.com/react/react_es6_spread.asp